### PR TITLE
feat(vendor-dev): use find-foreman for eslint extraneous dependencies

### PR DESCRIPTION
**Pull request by johnpmitsch**
This uses the `findForeman` package to check for dependencies in the eslint config. The benefits of this are:

- Always use the correct path in CI and locally
- Plugins can now use this configuration and all extraneous dependencies are covered in Foreman. `findForeman` will find foreman in a parent, child, or sibling (including itself so it works for foreman) directory. This means it will work in plugins as well as Foreman.

In Katello, we do [this now](https://github.com/Katello/katello/blob/f9a239d4e42ab95dcf2561f372df00d0cb47c42b/webpack/.eslintrc.js#L36). When this PR is added, we won't need to have that config added as it will already be done from extending this config. For plugins that extend this eslint config, the foreman dependencies will be considered for the extraneous dependencies check, meaning they don't have to add them to satisfy eslint rules.

This also adds the testing packages to be checked as dependencies. This makes it so testing packages don't have to be imported through @theforeman/test. With many packages available in the @theforeman/test, importing everything from @theforeman/test can obfuscate where the imports are coming from.

## Updated Packages

* [ ] root
* [ ] @theforeman/builder
* [ ] @theforeman/test
* [ ] @theforeman/eslint-plugin-foreman
* [ ] @theforeman/stories
* [ ] @theforeman/vendor
* [x] @theforeman/vendor-dev
* [ ] @theforeman/vendor-core
* [ ] @theforeman/find-foreman

## PR Type

* [ ] Bugfix
* [x] Feature
* [ ] Code style update (whitespace, formatting, missing semicolons, etc.)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Other… Please describe:

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Does this PR fixes open issues?

* [ ] Yes
* [x] No

## Do you use this PR in another repository?

* [x] Yes
* [ ] No

https://github.com/theforeman/foreman/pull/7807

### CODE REVIEW

File: packages/vendor-dev/eslint.extends.js [hunk 1]
```diff
@@ -12,14 +12,24 @@
...
+const foremanTestRelative = './node_modules/@theforeman/test/';
+const foremanTestDir = foremanRelativePath(foremanTestRelative);
+
 module.exports = {
   rules: {
     'import/no-extraneous-dependencies': [
       'error',
       {
-        packageDir: ['./', './node_modules/@theforeman/vendor-core/'],
+        packageDir: [foremanFull, foremanVendorDir, foremanTestDir],
```
Comment (sharvit) [inline packages/vendor-dev/eslint.extends.js:32]: Need to bring back the `'./',` so it would work for plugins

Comment (johnpmitsch) [reply packages/vendor-dev/eslint.extends.js]: @sharvit thanks, updated!

**Comment by johnpmitsch** [comment]
@sharvit looks like there was an issue deploying the release?

**Comment by sharvit** [comment]
> @sharvit looks like there was an issue deploying the release?

Right, see: https://travis-ci.org/github/theforeman/foreman-js/jobs/713272094#L325
I assume it is related to https://github.com/theforeman/foreman-js/commit/2ff8ac2650f634128d5e84aebfec143f13f865d9

Would you like to try fixing it? I would be able to merge it tomorrow/Sunday morning if works.

**Comment by johnpmitsch** [comment]
> > @sharvit looks like there was an issue deploying the release?
> 
> Right, see: https://travis-ci.org/github/theforeman/foreman-js/jobs/713272094#L325
> I assume it is related to [2ff8ac2](https://github.com/theforeman/foreman-js/commit/2ff8ac2650f634128d5e84aebfec143f13f865d9)
> 
> Would you like to try fixing it? I would be able to merge it tomorrow/Sunday morning if works.

I took a stab at it here https://github.com/theforeman/foreman-js/pull/183

I couldn't find what exactly changed in lerna that removed that option

**Pull request merged**